src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -22,38 +22,267 @@
+    # increment the num attempts
+    num_attempts = tf.add(num_attempts, 1.0)
+    aspect_ratio_var = random_ops.random_uniform([], min_aspect_ratio, max_aspect_ratio)
+    max_height_var = image_height
+    
+    max_height_from_width = tf.to_float(image_width) / aspect_ratio_var
+    height_condition = math_ops.greater(max_height_var, max_height_from_width)
+    max_height_var = control_flow_ops.cond(height_condition,
+                                    lambda: max_height_from_width,
+                                    lambda: max_height_var)
+    
+    max_height_from_area = tf.math.sqrt(max_area_fraction * image_height * image_width / aspect_ratio_var)
+    area_condition = math_ops.greater(max_height_var, max_height_from_area)
+    max_height_var = control_flow_ops.cond(area_condition,
+                                    lambda: max_height_from_area,
+                                    lambda: max_height_var)
+    
+    min_height_var = tf.math.sqrt(min_area_fraction * image_height * image_width / aspect_ratio_var)
+    crop_condition = math_ops.greater(min_height_var, max_height_var)
+    return control_flow_ops.cond(crop_condition,
+                                lambda: _crop_augmenter_loop_no_op(image_box, keep_annotations, annotations, num_attempts, max_attempts, perform_cropped_annotations, aspect_ratio_var, min_aspect_ratio, max_aspect_ratio, min_height_var, max_height_var, image_height, image_width, min_area_fraction, max_area_fraction),
+                                lambda: _crop_augmenter_loop_perform_crop(image_box, keep_annotations, annotations, num_attempts, max_attempts, perform_cropped_annotations, aspect_ratio_var, min_aspect_ratio, max_aspect_ratio, min_height_var, max_height_var, image_height, image_width, min_area_fraction, max_area_fraction))
+
+def _crop_augmenter_worker(image, annotation, num_attempts_var, max_attempts, perform_cropped_annotations, image_height, image_width, min_height_var, max_height_var, aspect_ratio_var, min_aspect_ratio, max_aspect_ratio, min_area_fraction, max_area_fraction, image_box, keep_annotations):
+    image_box, keep_annotations, annotation, num_attempts_var, max_attempts, perform_cropped_annotations, aspect_ratio_var, min_aspect_ratio, max_aspect_ratio, min_height_var, max_height_var, image_height, image_width, min_area_fraction, max_area_fraction = tf.while_loop(lambda image_box, keep_annotations, annotation, num_attempts_var, max_attempts, perform_cropped_annotations, aspect_ratio_var, min_aspect_ratio, max_aspect_ratio, min_height_var, max_height_var, image_height, image_width, min_area_fraction, max_area_fraction:_crop_augmenter_loop_conditional(num_attempts_var, max_attempts, perform_cropped_annotations),
```

Comment:
Does it need the identity lambda here, why not just pass in `_crop_augmenter_loop_conditional`?

Reply:
This is also almost impossible to read. Maybe some newlines?